File: UIViewPosition/ViewController.m
//
//  ViewController.m
//  UIViewPosition
//
//  Created by AbZorba Games on 11/04/2016.
//  Copyright © 2016 AbZorba Games. All rights reserved.
//

#import "ViewController.h"
#import "UIView+Position.h"

@interface ViewController ()
@property (nonatomic, strong) UIView *redView;
@property (nonatomic, strong) UIView *blueView;
@property (nonatomic, strong) UIView *greenView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _redView   = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    _blueView  = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    _greenView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];

    _redView.backgroundColor = [UIColor redColor];
    _blueView.backgroundColor = [UIColor blueColor];
    _greenView.backgroundColor = [UIColor greenColor];

    [self.view addSubview:_redView];
    [self.view addSubview:_blueView];
    [self.view addSubview:_greenView];
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    _redView.frameRightCenter = self.view.ownRightCenter;
    _blueView.frameRightCenter = _redView.frameBottomLeft;
    _greenView.frameBottomRight = _redView.frameTopCenter;

    [_redView floorFrame];
    [_blueView floorFrame];
    [_greenView floorFrame];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: UIViewPositionSwift/ViewController.swift
//
//  ViewController.swift
//  UIViewPositionSwift
//
//  Created by Georges Boumis on 12/04/2016.
//  Copyright © 2016 AbZorba Games. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    private var redView : UIView! = nil
    private var greenView : UIView! = nil
    private var blueView : UIView! = nil


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let origin = CGPoint.zero
        let size = CGSizeMake(100, 100)
        let frame = CGRect(origin: origin, size: size)
        redView = UIView(frame: frame)
        blueView = UIView(frame: frame)
        greenView = UIView(frame: frame)

        redView.backgroundColor = UIColor.redColor()
        blueView.backgroundColor = UIColor.blueColor()
        greenView.backgroundColor = UIColor.greenColor()

        self.view.addSubview(redView)
        self.view.addSubview(blueView)
        self.view.addSubview(greenView)
    }

    override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()

        redView.frameRightCenter   = self.view.ownRightCenter
        blueView.frameRightCenter  = redView.frameBottomLeft
        greenView.frameBottomRight = redView.frameTopCenter

        redView.floorFrame()
        blueView.floorFrame()
        greenView.floorFrame()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

